src/main/java/com/codessquad/qna/service/PostService.java

```diff
@@ -15,22 +16,39 @@
 public class PostService {
 
     private final PostRepository postRepository;
+    private final CommentRepostiory commentRepostiory;
 
-    @Autowired
-    public PostService(PostRepository postRepository) {
+    public PostService(PostRepository postRepository, CommentRepostiory commentRepostiory) {
         this.postRepository = postRepository;
+        this.commentRepostiory = commentRepostiory;
     }
 
     public void addPost(Post post) {
         postRepository.save(post);
     }
 
+    //Overload addPost
+    public void addPost(PostDto postDto) {
+        postRepository.save(Mapper.mapToPost(postDto));
+    }
+
     public Post getPost(Long id) {
         return postRepository.findById(id).orElseThrow(CanNotFindPostException::new);
     }
 
     public List<Post> getPosts() {
-        return postRepository.findAll();
+        return (List<Post>) postRepository.findAll();
```

Comment:
https://stackoverflow.com/questions/34702252/spring-data-jpa-crudrepository-returns-iterable-is-it-ok-to-cast-this-to-list
이 글 한 번 참고해보세요~

Reply:
**While a List is guaranteed to be an Iterable an Iterable may not be a List. This means that if you do cast an Iterable to a List it may fail at runtime. **

진짜 좋은 내용이네요!

Reply:
내부 동작가다보니 
```java
	@Override
	public Iterable<Object> invokeFindAll(Sort sort) {
		return customFindAllMethod ? super.invokeFindAll(sort) : repository.findAll();
	}
```

이런식으로 iterable 으로 넘겨주는데 추상화 단계라고 해야하나요..? 여하튼 윗단에서 이렇게 넘겨줄때
저희가 밑에 인터페이스에 특정 타입 List<> 등으로 제한 지어서 받는것이 훨씬 좋은 코드작성 방법인 것 같네요.
```java
 List<User> findAll();
```
만약에 위처럼 받으면 Iterable 같이 넓은 범위의 타입보다는 제한시킨 List 가 훨씬 좋다고 생각됩니다.
그래서 stackOverFlow 에서도 **This means that if you do cast an Iterable to a List it may fail at runtime.** 라고 적어준것 같다고 생각되네요.

혹시 이런 생각이 맞나요?

Reply:
불필요한 캐스팅이기도 하고, 충분한 공부가 되신 것 같은데요?